lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -2130,6 +2197,23 @@ void HdVP2Material::CompiledNetwork::Sync(
     HdSceneDelegate*            sceneDelegate,
     const HdMaterialNetworkMap& networkMap)
 {
+    auto updateShaderInstance = [this, &sceneDelegate](const HdMaterialNetwork& bxdfNet) {
+        const bool wasTransparent = _transparent;
+        _UpdateShaderInstance(sceneDelegate, bxdfNet);
+        // If the transparency flag changed, then the drawItems must be updated.
+        // e.g. if MRenderItem was first marked transparent and then the shader's transparency
+        // is turned off, MRenderItem must now be marked opaque.
+        bool drawItemsDirty = (wasTransparent != _transparent);
+
+// Consolidation workaround requires dirtying the mesh even on a ValueChanged
+#ifdef HDVP2_MATERIAL_CONSOLIDATION_UPDATE_WORKAROUND
+        drawItemsDirty = true;
+#endif
+        if (drawItemsDirty) {
+            _owner->MaterialChanged(sceneDelegate);
+        }
```

Comment:
I'm unsure whether this issue originates from maya or the vp2RenderDelegate. Currently, I trigger a resync when the transparency flag changes due to the existing implementation in HdVP2Mesh::_UpdateDrawItem.

https://github.com/Autodesk/maya-usd/blob/dae864e0b880ee0a19f1463c935632165f5274d2/lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp#L1815-L1816

https://github.com/Autodesk/maya-usd/blob/dae864e0b880ee0a19f1463c935632165f5274d2/lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp#L2269

So it marks MRenderItem as transparent if the mesh data includes opacity primvars OR if the shader is transparent. For example, if the shader is opaque at first sync, the render item is forced to be opaque. If the shader becomes transparent, the MRenderItem is still forced opacque and needs to be resynced.

I'm not sure why this approach is used. I would expect MRenderItem to automatically determine its transparency based on the shader instance, and only require `setTreatAsTransparent(true)` in case of other transparency source (e.g., per-vertex alphas). 

This might still be a VP2 issue or limitation. But in any case, logging this is not necessary on our side.